### ISSUE TITLE
BREAD relationships shouldn't use snake_case methods

### ISSUE THREAD

**Issue by Frondor** [thedevdojo/voyager#641]
Starting by the fact that [All methods should be camelCase per PSR-1.](https://github.com/laravel/framework/issues/4307#issuecomment-42037712), eloquent gives nightmares when dealing with model's attributes and relationship names.

Actually, I'm struggling with #638 because of this.
I suggest switching from relations like `user_id()` to simply `user()`.

Also, it would be nice to be able to specify foreign and primary keys.

**Comment by Frondor** [comment]
Finally, the solution I found for #638 PR was using `camelCase` relationships. If approved, the [documentation](https://the-control-group.github.io/voyager/docs/v0.10/#core-concepts-bread-validation) must be updated to 

> Note: the method used for this relationship, must match the camelCase version of the row's name from the pages table. Which is why we used authorId as the method name to tie the relationship.

**And all the relationships must have it's foreign key set.**

A proper example for an `user_id` (column name) relation:
### Row details:
```
{
    "relationship": {
        "key": "id",
        "label": "name",
        "page_slug": "admin/users"
    }
}
```
`page_slug` is optional

### Model method for this relationship
```
public function userId()
{
    return $this->(User::class, 'user_id');
}
```

**Comment by Frondor** [comment]
Merged #638
